### DISCUSSION TITLE
When unfolding code sections, previously folded sub-sections should stay folded

### DISCUSSION THREAD

**Discussion by tobydjones** [godotengine/godot-proposals#10671] [Editor]
I expect folded sub-sections of code to stay folded if I fold and unfold a higher level of indent, but they do not.

For example, if I have code with many layers of indentation, like this:
```
func message(A,B):
	match A:
		"GM":
			match B:
				"start_game":
					code...
				"stop_game":
					code...
		"PL":
			match B:
				"move":
					code...
				"shoot":
					code...
```
I can fold the lines "GM" and "PL", so I see this:
```
func message(A,B):
	match A:
		"GM": (...)
		"PL": (...)
```
If I then fold the function and unfold it again, I expect to see "GM" and "PL" folded, as in my second code example above, but it unfolds _every_ level, so I see _all_ the code, as in the top code example. This is frustrating when sub-sections such as "GM" and "PL" are several pages long - I end up having to re-fold each layer in turn to find the section of code I'm looking for.

Suggestions:

- Make folding permanent (until unfolded, obviously), perhaps as an option in the editor?
- Make "ALT-F" and clicking ">" fold & unfold only one level, preserving sub-level folds
- Add "ALT-SHIFT-F" and SHIFT-click ">" to fold & unfold that level and all sub-levels
- Make the menu command 'Fold all lines' fold all lines and not just the top level
- Make sure the menu command 'Unfold all lines' unfolds all lines that were previously folded
- Preserve folding when saving the file/project (perhaps in a project settings file?)

Thanks
Toby

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot/issues/82032

**Comment by Mickeon** [comment]
See also this:
- https://github.com/godotengine/godot/issues/85780

**Comment by Flowesans** [comment]
I almost create a new issue for this, and then I saw your discussion😭

**Reply by Calinou** [reply]
Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
